File: ios/vendored/sdk47/@stripe/stripe-react-native/ios/CardFormManager.m
#import <Foundation/Foundation.h>
#import <ABI47_0_0React/ABI47_0_0RCTBridgeModule.h>
#import <ABI47_0_0React/ABI47_0_0RCTViewManager.h>

@interface ABI47_0_0RCT_EXTERN_REMAP_MODULE(CardFormManager, ABI47_0_0CardFormManager, ABI47_0_0RCTViewManager)
ABI47_0_0RCT_EXPORT_VIEW_PROPERTY(onFormComplete, ABI47_0_0RCTDirectEventBlock)
ABI47_0_0RCT_EXPORT_VIEW_PROPERTY(dangerouslyGetFullCardDetails, BOOL)
ABI47_0_0RCT_EXPORT_VIEW_PROPERTY(autofocus, BOOL)
ABI47_0_0RCT_EXPORT_VIEW_PROPERTY(isUserInteractionEnabledValue, BOOL)
ABI47_0_0RCT_EXTERN_METHOD(focus:(nonnull NSNumber*) ABI47_0_0ReactTag)
ABI47_0_0RCT_EXTERN_METHOD(blur:(nonnull NSNumber*) ABI47_0_0ReactTag)
ABI47_0_0RCT_EXPORT_VIEW_PROPERTY(cardStyle, NSDictionary)
@end

File: ios/vendored/sdk47/@stripe/stripe-react-native/ios/CardFormManager.swift
import Foundation

@objc(ABI47_0_0CardFormManager)
class CardFormManager: ABI47_0_0RCTViewManager {
    override func view() -> UIView! {
        let cardForm = CardFormView()
        let stripeSdk = bridge.module(forName: "StripeSdk") as? StripeSdk
        stripeSdk?.cardFormView = cardForm;
        return cardForm
    }
    
    override class func requiresMainQueueSetup() -> Bool {
        return false
    }
    
    @objc func focus(_ abi47_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI47_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFormView = (viewRegistry![abi47_0_0ReactTag] as? CardFormView)!
            view.focus()
        }
    }
    
    @objc func blur(_ abi47_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI47_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFormView = (viewRegistry![abi47_0_0ReactTag] as? CardFormView)!
            view.blur()
        }
    }
}
